plugin/src/main/kotlin/org/gradle/github/dependency/extractor/ForceDependencyResolutionPlugin.kt

```diff
@@ -0,0 +1,30 @@
+package org.gradle.github.dependency.extractor
+
+import org.gradle.api.Plugin
+import org.gradle.api.invocation.Gradle
+
+@Suppress("unused")
+class ForceDependencyResolutionPlugin : Plugin<Gradle> {
+    override fun apply(gradle: Gradle) {
+        gradle.projectsEvaluated {
+            val resolveAllDeps = gradle.rootProject.tasks.register("dependencyExtractor_resolveBuildDependencies")
+
+            // Depend on "dependencies" task in all projects
+            gradle.allprojects { project ->
+                resolveAllDeps.configure {
+                    it.dependsOn(project.tasks.named("dependencies"))
+                }
+            }
+
+            // Depend on all 'resolveBuildDependencies' task in each included build
+            gradle.includedBuilds.forEach { includedBuild ->
+                resolveAllDeps.configure {
+                    it.dependsOn(includedBuild.task(":dependencyExtractor_resolveBuildDependencies"))
+                }
+            }
+
+            // Set the default task
+            gradle.rootProject.defaultTasks = listOf("dependencyExtractor_resolveBuildDependencies")
```

Comment:
I see that this sets the default task, but I'm not sure why. It might be would be worth it to expand on the comment explaining why the default task is being set. Does this impact builds which may already have `defaultTasks` set?

Reply:
I think I was being "cute" so that we could invoke Gradle without providing a task and it would automatically extract all dependencies. But you correctly point out that this would change behaviour for a build that already has `defaultTasks` set, and it isn't really necessary.

I'll change this so that we simply call `gradle dependencyExtractor_resolveBuildDependencies` when extracting the dependency graph. (I might want to rename the task since it will be more public).

Reply:
Done df3dd03